OsmAnd-telegram/src/net/osmand/telegram/helpers/LocationMessages.kt

```diff
@@ -140,8 +147,27 @@ class LocationMessages(val app: TelegramApplication) {
 		dbHelper.removeBufferedMessage(message)
 	}
 
+	private fun removeOldBufferedMessages() {
+		if (this.bufferedMessages.isNotEmpty()) {
```

Comment:
Request not more often than 1 minute. It also could simplified using .filter. Copy the list, retrieve only expired messages using filter and delete them from db and cache. Use mutableListOf<T>().apply { addAll(this) } for creating mutable copy. Also it would be good to replace mutableListOf(*this.bufferedMessages.toTypedArray()) with addAll.